tox.ini

```diff
@@ -1,5 +1,5 @@
 [tox]
-envlist = flake8,py37,py38,py39,py310
+envlist = flake8,py37,py38,py39,py310,py311,py312
```

Comment:
Below instead of installing 
`pip install . .[avro] .[schema-registry] .[json] .[protobuf]`
we can install
`pip install .[tests]`

Reply:
deps can be removed
```
deps =
    # https://docs.pytest.org/en/latest/changelog.html#id53
    -rrequirements/requirement-dev.txt
```

Reply:
Please add `,tmp-KafkaCluster/*` to the flake8 exclusion list, folder created by trivup

Reply:
Instead of `pip install .[tests]`, I have added another block for `src` and added `pip install .[src]` instead as there are some extra dependencies in tests which are not main dependencies. Refer pyproject.toml.

Reply:
But here it's a test so it's correct to use tests, if you don't it says that `pytest` isn't found (you need to remove the .tox folder first).

Reply:
Could you also remove
```
setenv =
    CPPFLAGS=-I{toxinidir}/tmp-build/include
    LDFLAGS=-L{toxinidir}/tmp-build/lib
    C_INCLUDE_PATH={toxinidir}/tmp-build/include
    LD_LIBRARY_PATH={toxinidir}/tmp-build/lib
``` 
It overwrites the env variables, for example I've set `C_INCLUDE_PATH` and `LD_LIBRARY_PATH` so it uses librdkafka folder instead of installing with `make install` every time. And this way it doesn't use the defined variables.